primitives/constraints/ssz.go

```diff
@@ -38,13 +38,6 @@ type SSZUnmarshaler interface {
 	EnsureSyntaxFromSSZ() error // once unmarshalled we will check whether type syntax is correct
 }
 
-// sszVersionedUnmarshaler is an interface for objects that can be
-// unmarshaled from SSZ format for specific versions.
-type sszVersionedUnmarshaler[SelfT any] interface {
-	// NewFromSSZ creates a new object from SSZ format with the given version.
-	NewFromSSZ(bz []byte, version common.Version) (SelfT, error)
-}
```

Comment:
once New is moved out of the interface, `sszVersionedUnmarshaler` becomes the same as `SSZUnmarshaller` to merged the interfaces